From 6b63299855e590250a33d2035c84b525e2b8f349 Mon Sep 17 00:00:00 2001
From: Naoto Yamaguchi <naoto.yamaguchi@aisin.co.jp>
Date: Wed, 10 Jun 2026 23:37:21 +0900
Subject: [PATCH] Fix build error

Upstream-Status: Inappropriate [embedded specific]

Signed-off-by: Naoto Yamaguchi <naoto.yamaguchi@aisin.co.jp>
---
 src/lxc/open_utils.h | 14 --------------
 1 file changed, 14 deletions(-)

diff --git a/src/lxc/open_utils.h b/src/lxc/open_utils.h
index 7ff5945c7..a683302be 100644
--- a/src/lxc/open_utils.h
+++ b/src/lxc/open_utils.h
@@ -19,13 +19,6 @@
  * @mode: O_CREAT/O_TMPFILE file mode.
  * @resolve: RESOLVE_* flags.
  */
-#if !HAVE_STRUCT_OPEN_HOW
-struct open_how {
-	__u64 flags;
-	__u64 mode;
-	__u64 resolve;
-};
-#endif
 
 /* how->resolve flags for openat2(2). */
 #ifndef RESOLVE_NO_XDEV
@@ -76,12 +69,5 @@ struct open_how {
 #define PROTECT_OPEN_W (PROTECT_OPEN_W_WITH_TRAILING_SYMLINKS | O_NOFOLLOW)
 #define PROTECT_OPEN_RW (O_CLOEXEC | O_NOCTTY | O_RDWR | O_NOFOLLOW)
 
-#if !HAVE_OPENAT2
-static inline int openat2(int dfd, const char *filename, struct open_how *how, size_t size)
-{
-	return syscall(__NR_openat2, dfd, filename, how, size);
-}
-#endif /* HAVE_OPENAT2 */
-
 #endif /* __LXC_OPEN_UTILS_H */
 
-- 
2.43.0

